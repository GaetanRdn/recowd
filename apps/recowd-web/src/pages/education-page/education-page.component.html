<rc-education-cards-list [educations]="educations"></rc-education-cards-list>
