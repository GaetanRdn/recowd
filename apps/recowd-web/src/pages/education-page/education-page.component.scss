:host {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;

  rc-education-cards-list {
    width: clamp(42.5rem, 60.2rem, 100%);
  }
}
