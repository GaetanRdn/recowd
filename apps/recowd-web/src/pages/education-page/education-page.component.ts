import { ChangeDetectionStrategy, Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Education } from '@recowd/models';
import { EducationCardComponent } from '@recowd/ui/molecules';
import { AvatarDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-education-page',
  standalone: true,
  imports: [CommonModule, EducationCardComponent, AvatarDirective],
  templateUrl: './education-page.component.html',
  styleUrls: ['./education-page.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class EducationPageComponent {
  protected readonly educations: Education[] = [
    {
      degree: 'Master MIAGE par apprentissage',
      from: 2012,
      to: 2014,
      description:
        "Université Paris-Sud, Méthodes Informatique Appliquées à la Gestion d'entreprise",
      logo: 'universite-paris-sud.jpeg',
    },
    {
      degree: 'Licence MIAGE par apprentissage',
      from: 2011,
      to: 2012,
      description:
        "Université Paris-Sud, Méthodes Informatique Appliquées à la Gestion d'entreprise",
      logo: 'universite-paris-sud.jpeg',
    },
    {
      degree: 'DUT informatique et gestion',
      from: 2008,
      to: 2010,
      description: "IUT Aix-Marseille II, informatique et gestion d'entreprise",
      logo: 'iut-aix-marseille.jpeg',
    },
  ];
}
