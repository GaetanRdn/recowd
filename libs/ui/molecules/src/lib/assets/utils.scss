@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
