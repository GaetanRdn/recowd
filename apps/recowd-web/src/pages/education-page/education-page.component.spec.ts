import { TestBed, waitForAsync } from '@angular/core/testing';
import { EducationPageComponent } from './education-page.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('EducationPageComponent', () => {
  let templateLookup: TemplateLookup<EducationPageComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [EducationPageComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(EducationPageComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
