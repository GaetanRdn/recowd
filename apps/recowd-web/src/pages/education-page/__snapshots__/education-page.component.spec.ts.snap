// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationPageComponent should create 1`] = `
HTMLCollection [
  <rc-education-cards-list>
    <rc-education-card>
      <rc-card>
        <div
          class="rc-card-header"
        >
          
          <rc-card-title
            class="rc-card-title"
          >
            Master MIAGE par apprentissage
          </rc-card-title>
          <rc-card-subtitle
            class="rc-card-subtitle"
          >
            2012 - 2014
          </rc-card-subtitle>
        </div>
        <rc-card-content
          class="rc-card-content"
        >
          Université Paris-Sud, Méthodes Informatique Appliquées à la Gestion d'entreprise
        </rc-card-content>
      </rc-card>
    </rc-education-card>
    <rc-education-card>
      <rc-card>
        <div
          class="rc-card-header"
        >
          
          <rc-card-title
            class="rc-card-title"
          >
            Licence MIAGE par apprentissage
          </rc-card-title>
          <rc-card-subtitle
            class="rc-card-subtitle"
          >
            2011 - 2012
          </rc-card-subtitle>
        </div>
        <rc-card-content
          class="rc-card-content"
        >
          Université Paris-Sud, Méthodes Informatique Appliquées à la Gestion d'entreprise
        </rc-card-content>
      </rc-card>
    </rc-education-card>
    
  </rc-education-cards-list>,
]
`;
